src/services/qrService.ts

```diff
@@ -277,17 +276,16 @@ const connectOpenId4VcIssuance = async (args: IQrDataArgs) => {
       })
 
   const provider = await OpenId4VcIssuanceProvider.initiationFromUri({ uri: args.qrData.uri })
-  provider.getServerMetadataAndPerformCryptoMatching().then((metadata: IServerMetadataAndCryptoMatchingResponse) => {
+  provider.getServerMetadataAndPerformCryptoMatching().then(async (metadata: IServerMetadataAndCryptoMatchingResponse) => {
     const url = new URL(metadata.serverMetadata.issuer)
-    // TODO WAL-380 should be handled by the connection-manager
-    if (!store.getState().contact.contacts.some((contact: IConnectionParty) => contact.name === url.host)) {
-      // TODO fix this type issue
+    const contacts = await getContacts({ filter: [{ name: url.host }] })
```

Comment:
This should become uri as key with a value of url.protocol and hostname instead of name right? I know the user cannot change the name right now. That is what we have the alias for. But since this is the contact we are setting the uri to the generic hostname already.

The problem is that the user could update the uri I guess. Maybe we should add a baseUri or initialURI or something

Reply:
uri is optional as well, so i dont tthink that will work